contributing_guidelines.md

```diff
@@ -0,0 +1,191 @@
+* ...
+
+Any Java package MUST match a couple of paths like these ones:
+
+* `/<agent_name>/src/main/java/com/telefonica/iot/cygnus/<etc>/<etc>/…` → functional code
+* `/<agent_name>/src/test/java/com/telefonica/iot/cygnus/<etc>/<etc>/…` → unit tests
+
+As can be seen, despite the repository organization, from a Java perspective all the agents code is under `com.telefonica.iot.cygnus`.
+
+[Top](#top)
+
+##<a name="section5"></a>Backlog
+The <i>issues</i> section of the main repository MUST be used for tracking all the features, hardening, bugs and task to be implemented by every agent.
+
+The name of each issue MUST follow the following format:
+
+    [<agent name>] [feature|hardening|bug|task] <short description>
+
+Where <i>short description</i> MAY enclose other “[...]” sublevels. For instance:
+
+    [cygnus-ngsi] [hardening] [grouping rules] Precompile patterns from regexes
+
+Alternatively, labels for each agent and task type SHOULD be created.
+
+Every issue MUST have a description as detailed as the creator considers, but it MUST be enough to understand the purpose of the issue and to allow the community starts a discussion.
+
+Every issue SHOULD have an associated sprint/milestone among the ones in the <i>milestonres</i> section on the main repository.
+
+There MUST NOT be assignee because each issue is considered to be assigned to a development team related to the agent; so, the assignation MUST be done internally to the team. Anyway, the real Github user ID assignee to the issue MAY be added to the description of the issue; in that case, the following format MUST be used:
+
+    * Assignee: @<Github’s user ID>
+
+[Top](#top)
+
+##<a name="section6"></a>Contributing to the repository
+###<a name="section6.1"></a>Main repository versus forked repositories
+Every team in charge of an agent MUST create one or more forks of the main repository. Every team SHOULD synchronize their forked repositories with the main one after opening a pull request (see next section).
+
+Only those contributions merged into the main repository MUST be considered as part of the official Cygnus development.
+
+[Top](#top)
+
+###<a name="section6.2"></a>Pull requests
+Any contribution MUST be done through a new opened pull request (PR). These PRs MUST compare certain branch at any forked repository against the `develop` base branch in the main repository.
+
+At least one people from Telefónica’s team MUST participate in the PR review, and MUST agree with the PR before merging it. The agreement MAY be expressed with the “LGTM” (looks good to me) keyword. In order to agilize the PR process, this kind of review MUST be exclusively related to the alignment of the PR with these contribution guidelines. Any comment regarding the content MAY be added anyway.
+
+When a PR regarding the `cygnus-common` folder is done, then the review MUST agree both with the guidelines fulfillment and the content.
+
```

Comment:
However, cygnus-common contributions must be fully reviewed and approved by a member of the Cygnus Core Team.